src/google-maps/map-polyline/map-polyline.ts

```diff
@@ -0,0 +1,234 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+// Workaround for: https://github.com/bazelbuild/rules_nodejs/issues/1265
+/// <reference types="googlemaps" />
+
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  OnDestroy,
+  OnInit,
+  Output,
+  ViewEncapsulation
+} from '@angular/core';
+import {BehaviorSubject, combineLatest, Observable, Subject} from 'rxjs';
+import {map, take, takeUntil} from 'rxjs/operators';
+
+import {GoogleMap} from '../google-map/google-map';
+
+/**
+ * Angular component that renders a Google Maps Polyline via the Google Maps JavaScript API.
+ * @see developers.google.com/maps/documentation/javascript/reference/polygon#Polyline
+ */
+@Component({
+  moduleId: module.id,
+  selector: 'map-polyline',
+  template: '<ng-content></ng-content>',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+})
+export class MapPolyline implements OnInit, OnDestroy {
```

Comment:
Shouldn't we call this a `polygon` to align it with the naming in the Google Maps API?

Reply:
Polyline is actually separate from Polygon: https://developers.google.com/maps/documentation/javascript/reference/polygon#Polyline